community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java

```diff
@@ -382,8 +390,14 @@ public static void main( String[] incomingArguments ) throws IOException
      */
     public static void main( String[] incomingArguments, boolean defaultSettingsSuitableForTests ) throws IOException
     {
-        System.err.println("WARNING: neo4j-import is deprecated and support for it will be removed in a future\n" +
-                "version of Neo4j; please use neo4j-admin import instead.\n");
+        System.err.println( format( "WARNING: neo4j-import is deprecated and support for it will be removed in a future%n" +
+                "version of Neo4j; please use neo4j-admin import instead." ) );
+        if ( Runtime.getRuntime().maxMemory() > RECOMMENDED_MAX_HEAP_SIZE )
```

Comment:
those recommendations should be correlated with the recommendation that memrec provides

Reply:
You mean some official memory recommendations? Where are those? Also, this is different, import needs virtually no heap at all... typically only max a couple of gigs. For very large stores the page cache meta data will take up a big chunk of the heap tho so therefore this is a bit higher. The limit is meant to be set to 32GiB, but being on the safe side such that object pointers can still fit in 4 bytes. If that limit is optimally 29GiB, so be it, I can change it, but I don't think it matters much. 28-29 GiB is already too high for an import, unless it's in the several hundred billions.

Reply:
i would assume you referring to https://wiki.openjdk.java.net/display/HotSpot/CompressedOops and you want to keep oops to be 32 bits and never cross that boundary right? exactly with that limitations/suggestions in mind our recent PR https://github.com/neo4j/neo4j/pull/10821 will provide some recommended heap settings. And memrec that I'm referring to in the above comment is exactly that tool.
The problem here is different recommendations that popping up from different tools that misguide and confuse users. Would be really nice to avoid that.

Reply:
Yes, that's what I'm referring to.

`memrec` doesn't apply here because the use case is so different from running a database. E.g. in importer the page cache has an optimal size of 500MiB - 1GiB and heap around a couple of gigabytes for most datasets, virtually regardless of your machine spec. For really huge datasets the page cache still has the same recommended size and heap a little higher.